examples/nlp/character_and_byte_level_text_generation.py

```diff
@@ -0,0 +1,285 @@
+
+"""
+KerasNLP provides lots of building blocks for NLP (model layers, tokenizers, metrics, etc.) and
+makes it convenient to construct NLP pipelines on the fly.
+
+In this tutorial we'll use KerasNLP's `UnicodeCharacterTokenizer` and `ByteTokenizer` to train a 
+character level LSTM model for open ended text generation. This example draws inspiration from 
+[Character-level recurrent sequence-to-sequence model example](https://keras.io/examples/generative/lstm_character_level_text_generation/) 
+by [fchollet](https://twitter.com/fchollet) for it's model architecture and uses 
+[Don Quijote by Miguel de Cervantes Saavedra (Spanish Version)](https://www.gutenberg.org/ebooks/2000) as a data source
+
+This tutorial broadly covers the following:
+- Tokenization using `keras_nlp.tokenizers.UnicodeCharacterTokenizer` and `keras_nlp.tokenizers.ByteTokenizer` to obtain character level tokens.
+- A LSTM model to generate text character-by-character 
+- Decoding using `keras_nlp.utils.greedy_search` utility
+
+This tutorial will be pretty useful and will be a good starting point for learning about KerasNLP and how to 
+incorporate it into your own NLP pipelines.
+"""
+
+"""
+## Setup
+
+Let's do our imports first
+"""
+
+import tensorflow as tf
+from tensorflow import keras
+from keras import layers
+import keras_nlp
+import numpy as np
+import random
+import io
+
+"""
+Let's get our dataset and do some preprocessing to remove new line characters and the meta data present in the text
+"""
+
+path = keras.utils.get_file(
+    "spanish_version.txt", origin="https://www.gutenberg.org/files/2000/2000-0.txt"
+)
+with io.open(path, encoding="utf-8") as f:
+    text = f.read()
+text = text.replace("\n", " ")
+
+starting_text = "*** start of the project gutenberg ebook don quijote ***"
+ind_start = text.find(starting_text)
+ending_text = "*** end of the project gutenberg ebook don quijote ***"
+ind_end = text.find(ending_text)
+text = text[ind_start + len(starting_text):ind_end]
+text = ' '.join(text.split())
+
+"""
+Now we'll create out tokenizers and tokenize our text, we limit the vocabulary for the UnicodeCharacterTokenizer to 
+the max value in the Latin Extended-B block of Unicode version 1.1 as any thing beyond that is not Spanish text and 
+probably should not be in our dataset
+"""
+
+UnicodeCharacterTokenizer = keras_nlp.tokenizers.UnicodeCharacterTokenizer(vocabulary_size= 591)
+ByteTokenizer = keras_nlp.tokenizers.ByteTokenizer()
+
+unicode_tokenized_text = UnicodeCharacterTokenizer(text)
+byte_tokenized_text = ByteTokenizer(text)
+
+"""
+Here's what our tokens look like
+"""
+
+print(unicode_tokenized_text[:50])
+print(byte_tokenized_text[:50])
+
+"""
+Let's see how we can detokenize the text!
+"""
+
+print(UnicodeCharacterTokenizer.detokenize(unicode_tokenized_text[:50]))
+print(ByteTokenizer.detokenize(byte_tokenized_text[:50]))
+
+"""
+We can see post tokenization the datasets now differ which is to be expected as byte tokenization and unicode 
```

Comment:
maybe just phrase this as

Let's try tokenizing the the input string "señora mía".

print(codepoint_tokenizer("señora mía))
print(byte_tokenizer("señora mía))

This captures the tradeoff between the two tokenizer. The byte tokenizer will handle any text with only 256 output ids, at the cost of encoding to a longer sequence. The unicode tokenizer will produce shorted sequence, at the cost of a larger id space (and larger embeddings).